src/averageAges.js

```diff
@@ -16,10 +16,13 @@
  */
 function calculateMenAverageAge(people, century) {
   // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(p => p.sex === 'm');
+  const selectedMen = century
+    ? men.filter(p => Math.ceil(p.died / 100) === century)
+    : men;
+
+  return selectedMen.map(p => p.died - p.born)
+    .reduce((a, b) => a + b, 0) / selectedMen.length;
```

Comment:
a helper function to calculate averages would be really helpful

Reply:
I added :)